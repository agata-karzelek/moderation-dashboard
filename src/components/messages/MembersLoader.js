import React from "react";
import { Waypoint } from "react-waypoint";
import { Grid, CircularProgress } from "@material-ui/core";

export default function MembersLoader(props) {
  return (
    <>
      {props.membersLength > 99 && props.membersLength < props.totalCount ? (
        <Waypoint onEnter={props.fetchChannelsMembersOnScroll} />
      ) : null}
      {props.loading ? (
        <Grid justify="center" container>
          <CircularProgress />
        </Grid>
      ) : null}
    </>
  );
}
