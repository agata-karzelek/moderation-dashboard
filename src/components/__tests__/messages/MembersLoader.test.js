import React from "react";
import { shallow } from "enzyme";
import MembersLoader from "../../messages/MembersLoader";

describe("Test Case For Members Loader", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <MembersLoader
        loading={false}
        membersLength={97}
        fetchChannelsMembersOnScroll={jest.fn()}
        totalCount={190}
      />
    );
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
