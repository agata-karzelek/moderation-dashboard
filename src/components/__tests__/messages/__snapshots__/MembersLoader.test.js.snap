// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For Members Loader Snapshot 1`] = `<Fragment />`;
